### PR TITLE
feat: add disable client setinfo feature and its default mode is off

### LINKED ISSUE #1034: Opt-out client set info when creating new connection
https://github.com/redis-rs/redis-rs/blob/main/redis/src/connection.rs#L933

`CLIENT SET-INFO` will be called in a pipeline twice when creating a new connection, and this will takes about 2-4ms to complete in our environment.

Redis is commonly used for low latency stuffs and this can lead to a significant latency increase especially when retry is added, which greatly impacts the pct99 latency(the p99 latency of our service before adding this behavior is only about 2-3ms).

Can this be designed as an option or something that can be opt-out by feature(such as the `tcp_nodelay` feature)?

This is important for us. Thanks!

### PR DESCRIPTION
**Pull request by Ggiggle**
fix #1034

### CODE REVIEW

**Comment by nihohit** [comment]
also https://github.com/redis-rs/redis-rs/issues/1019

File: redis/Cargo.toml [hunk 1]
```diff
@@ -133,7 +158,12 @@ futures = "0.3"
 criterion = "0.4"
 partial-io = { version = "0.5", features = ["tokio", "quickcheck1"] }
 quickcheck = "1.0.3"
-tokio = { version = "1", features = ["rt", "macros", "rt-multi-thread", "time"] }
+tokio = { version = "1", features = [
+    "rt",
+    "macros",
+    "rt-multi-thread",
+    "time",
+] }
```
Comment (nihohit) [inline redis/Cargo.toml:166]: please revert these style changes.

Comment (Ggiggle) [reply redis/Cargo.toml:166]: Okay.

**Review by nihohit — APPROVED** [review]

**Comment by Ggiggle** [comment]
The failed tests seem not relative with my pr, please take a look at this error.

**Comment by nihohit** [comment]
> The failed tests seem not relative with my pr, please take a look at this error.

it isn't. Thanks for your contribution!

**Pull request merged**